src/libstd/thread/mod.rs

```diff
@@ -610,27 +594,19 @@ impl thread_info::NewThread for Thread {
 #[stable(feature = "rust1", since = "1.0.0")]
 pub type Result<T> = ::result::Result<T, Box<Any + Send + 'static>>;
 
-struct Packet<T>(Arc<UnsafeCell<Option<Result<T>>>>);
-
-unsafe impl<T:Send> Send for Packet<T> {}
-unsafe impl<T> Sync for Packet<T> {}
+type Packet<T> = Arc<Mutex<Option<Result<T>>>>;
```

Comment:
This helped avoid a number of `unsafe impl` blocks as well as removing a bit of `unsafe` code. I definitely agree that it was properly synchronized before, I was just hoping to cut down on the amount of `unsafe` here. I'll see if I can't finesse a nicer solution to this though.